### ISSUE TITLE
[Chore] Extract business logic from controller 

### ISSUE THREAD

**Issue by olivierobert** [buigiaanfb1/extract-data#21]
## Issue

In the current implementation, the controllers contain all the business logic (e.g., DB queries, scraping), making it hard to test and maintain.

## Expected

The controllers should be limited to routing concerns and return the adequate response content and HTTP code. All business logic should be extracted into repository objects (for DB operations) and services (or also commonly used cases). These plain JS objects

**Comment by buigiaanfb1** [comment]
@olivierobert yeah, thank you for pointing it out. I will create another folder to handle the business logic to make it easy to read and maintain
